staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -86,13 +91,20 @@ type watchChan struct {
 	errChan           chan error
 }
 
-func newWatcher(client *clientv3.Client, codec runtime.Codec, versioner storage.Versioner, transformer value.Transformer) *watcher {
-	return &watcher{
+func newWatcher(client *clientv3.Client, codec runtime.Codec, newFunc func() runtime.Object, versioner storage.Versioner, transformer value.Transformer) *watcher {
```

Comment:
Unit tests are still missing for changes in this file, but they are blocked on our ability to tun etcd for testing:
https://github.com/etcd-io/etcd/issues/12265

Will work on making that happen.

Reply:
The tests are now added.